File: 백준/Silver/16953. A → B/A → B.py
from collections import deque
import sys

input = sys.stdin.readline

a, b = map(int, input().split())

answer = []
def dfs(current, target, cnt):
    if current == target:
        answer.append(cnt)
        return
    if current > target:
        return
    dfs(current * 2, target, cnt + 1)
    dfs(current * 10 + 1, target, cnt + 1)
        
dfs(a, b, 0)

if answer:
    print(min(answer) + 1)
else:
    print(-1)

File: 백준/Silver/16953. A → B/A → B.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let lines = [];

rl.on("line", (input) => {
  lines.push(input.trim());
  if (lines.length === 1) {
    rl.close();
  }
});

rl.on("close", () => {
  const [a, b] = lines[0].split(" ").map(Number);

  // 2를 곱한다.
  // 1을 맨 뒤에 추가한다.

  const answer = [];
  function dfs(num, cnt) {
    if (num === b) {
      answer.push(cnt);
      return;
    }
    if (num > b) return;
    dfs(num * 2, cnt + 1);
    dfs(num * 10 + 1, cnt + 1);
  }

  dfs(a, 0);
  if (answer.length) {
    console.log(Math.min(...answer) + 1);
  } else {
    console.log(-1);
  }
});
